src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -523,7 +523,20 @@ public function cleanup(): void
 
     public function getMessageCount(): int
     {
-        return $this->connection->xlen($this->stream);
+        $groups = $this->connection->xinfo('GROUPS', $this->stream);
+
+        if (false === $groups) {
+            return 0;
+        }
+
+        foreach ($groups as $group) {
+            if ($group['name'] === $this->group) {
+                $lastDeliveredID = $group['last-delivered-id'];
+                break;
+            }
+        }
```

Comment:
```suggestion
        $lastDeliveredID = null;
        foreach ($groups as $group) {
            if ($group['name'] === $this->group) {
                $lastDeliveredID = $group['last-delivered-id'];
                break;
            }
        }
        
        if ($lastDeliveredID === null) {
            return 0;
        }
```

it could be possible that the `last-delivered-id` is not set.

Reply:
That's why I used `$lastDeliveredID ?? '0-0'` in `xrange` call. It should be enough, right ?

Reply:
I did go more in the case were xinfo does never have the `last-delivered-id` set in that case we can not call `xRange` on `0-0` as it would then return a false value. I think that would be the case for all php redis extension from version ^5.0. but not sure.

Reply:
OK. Change made.